### PR TITLE
EZ-678: Customize quick links to add my own shortcut links

### PR DESCRIPTION
**Pull request by saurabh-2k17**

### CODE REVIEW

File: ezcontent_publish.module [hunk 1]
```diff
@@ -60,3 +64,73 @@ function ezcontent_publish_entity_insert(EntityInterface $entity) {
...
+    'usershortcutblock_template' => [
+      'variables' => ['data' => NULL],
+    ]
+  ];
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function ezcontent_publish_form_shortcut_set_customize_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
```
Comment (arshadkhan53) [inline ezcontent_publish.module]: we can add this in use statement `\Drupal\Core\Form\FormStateInterface`

**Review by arshadkhan53 — DISMISSED** [review]

File: ezcontent_publish.module [hunk 2]
```diff
@@ -60,3 +64,73 @@ function ezcontent_publish_entity_insert(EntityInterface $entity) {
...
+ */
+function ezcontent_publish_shortcut_form_submit(array $form, \Drupal\Core\Form\FormStateInterface $form_state){
+  Cache::invalidateTags(['ezcontent_user_shortcut_block']);
+  $shortcutSet = $form_state->getBuildInfo()['callback_object']->getEntity()->get('shortcut_set')->getValue()[0]['target_id'];
+  $redirectUrl = Url::fromRoute('ezcontent_publish.shortcut_set.customize_form', array('shortcut_set' => $shortcutSet));
+  $form_state->setRedirectUrl($redirectUrl);
+}
+
+/**
+ * Implements get_usershortcut_links().
```
Comment (arshadkhan53) [inline ezcontent_publish.module]: Hey @saurabh-2k17 can we move this function to block itself.

File: ezcontent_publish.module [hunk 3]
```diff
@@ -60,3 +64,73 @@ function ezcontent_publish_entity_insert(EntityInterface $entity) {
...
   }
 }
+
+/**
+ * Implements hook_theme().
+ *
+ */
+function ezcontent_publish_theme($existing, $type, $theme, $path) {
+  return [
+    'usershortcutblock_template' => [
```
Comment (arshadkhan53) [inline ezcontent_publish.module]:
```suggestion
    'user_shortcut_block_template' => [
```

File: ezcontent_publish.module [hunk 4]
```diff
@@ -60,3 +64,73 @@ function ezcontent_publish_entity_insert(EntityInterface $entity) {
...
+  $operationShortcutLink = Link::fromTextAndUrl(t('Visit Shortcuts'), $operationShortcutUrl)->toRenderable();
+  // Loading current user shortcuts for display.
+  $currentUserShortcuts =  \Drupal::entityTypeManager()
+    ->getStorage('shortcut')
+    ->loadByProperties(['shortcut_set' => $currentUserShortcutSet]);
+  foreach ($currentUserShortcuts as $key => $currentUserShortcutObj) {
+    $shortcutTitle = $currentUserShortcutObj->get('title')->value;
+    $shortcutLink =$currentUserShortcutObj->get('link')->first()->getUrl();
+    $shortcutLink = Link::fromTextAndUrl($shortcutTitle, $shortcutLink)->toRenderable();
+    $shortcutObj['links'][] = $shortcutLink;
```
Comment (arshadkhan53) [inline ezcontent_publish.module]:
```suggestion
    $shortcutObj['links'][] = Link::fromTextAndUrl($shortcutTitle, $shortcutLink)->toRenderable();
```

**Review by arshadkhan53 — CHANGES_REQUESTED** [review]

**Comment by saurabh-2k17** [comment]
This file has been covered in https://github.com/srijanone/ezcontent_publish/pull/42 this PR. Hence, closing this one.

**Pull request closed without merging**